components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/service/ActionPropertyResolver.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management.service;
+
+import org.wso2.carbon.identity.action.management.exception.ActionPropertyResolverException;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.ActionDTO;
+
+import java.util.List;
+
+/**
+ * This interface defines the Action Property Resolver.
+ * Action Property Resolver is the component that is responsible for handling action type specific operations.
+ */
+public interface ActionPropertyResolver {
+
+    Action.ActionTypes getSupportedActionType();
+
+    /**
+     * Resolve the properties that need to be added in the Action Management Service.
+     * This method is responsible for performing necessary CRUD operations for the properties that need to be added
+     * using other external services.
+     * The Action Management Service ensures that only the properties returned by this method are stored.
+     *
+     * @param actionDTO    ActionDTO object.
+     * @param tenantDomain Tenant domain.
+     * @return ActionDTO object with resolved properties.
+     * @throws ActionPropertyResolverException If an error occurs while resolving the properties.
+     */
+    default ActionDTO resolveAddingProperties(ActionDTO actionDTO, String tenantDomain)
```

Comment:
Shall we add a better name.
Like populatePropertiesForAddOperation, populateDTOForAddOperation likewise

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6119/commits/2d1289a369498b88e58048181ecc730cf3265997